File: Golang/running_sum_of_1d_array.go
package main

func RunningSum(nums []int) []int { // [1,2,3,4]
	var result []int

	for i := 0; i < len(nums); i++ {
		if i == 0 {
			result = append(result, nums[i])
			continue
		}

		var temp int = 0
		for _, n := range nums[:i + 1] {
			temp += n
		}
		result = append(result, temp)
	}

	return result
}

File: Javascript/runningSumOf1dArray.js
var runningSum = function(nums) {
    const result = new Array(nums.length);

  
    result[0] = nums[0];  // elemen pertama

    for (let i = 1; i < nums.length; i++) {
        result[i] = result[i - 1] + nums[i]; // akumulasi dari nilai sebelumnya
    }

    return result;
};

console.log(runningSum([1, 2, 3, 4]));
